lib/rules/no-unused-styles.js

```diff
@@ -1,70 +1,29 @@
-/**
- * @fileoverview Detects unused styles
- * @author Tom Hastjarjanto
- */
-
 'use strict';
 
-const Components = require('../util/Components');
-const styleSheet = require('../util/stylesheet');
-
-const { StyleSheets } = styleSheet;
-const { astHelpers } = styleSheet;
+const noUnusedStylesStable = require('./no-unused-styles.stable');
 
-const create = Components.detect((context, components) => {
-  const styleSheets = new StyleSheets();
-  const styleReferences = new Set();
+function create(context) {
+  const isExperimental = context.options[0] && context.options[0].enableImportsCheck === true;
 
-  function reportUnusedStyles(unusedStyles) {
-    Object.keys(unusedStyles).forEach((key) => {
-      if ({}.hasOwnProperty.call(unusedStyles, key)) {
-        const styles = unusedStyles[key];
-        styles.forEach((node) => {
-          const message = [
-            'Unused style detected: ',
-            key,
-            '.',
-            node.key.name,
-          ].join('');
+  if (isExperimental) {
```

Comment:
what exactly is experimental about this feature? In what cases will it not work as expected?

Reply:
I don't want to break back compatibility. As I add so complex logic and it can include some unexpected behavior + impact on performance

Better hide it behind bool flag